shared/parcel.js

```diff
@@ -171,28 +44,30 @@ export function normalizeParcel(parcel, prevParcel = {}) {
 
 export function connectParcels(array, parcels) {
   array.forEach(parcel => {
-    const { x, y } = parcel
-    if (parcels[parcel.id].district_id != null) {
+    const { id, x, y } = parcel
+    if (parcels[id].in_estate || parcels[id].district_id != null) {
       const leftId = buildCoordinate(x - 1, y)
       const topId = buildCoordinate(x, y + 1)
       const topLeftId = buildCoordinate(x - 1, y + 1)
 
-      parcels[parcel.id].connectedLeft =
-        !parcels[leftId] ||
-        isSameValue(parcels[parcel.id], parcels[leftId], 'district_id')
-
-      parcels[parcel.id].connectedTop =
-        !parcels[topId] ||
-        isSameValue(parcels[parcel.id], parcels[topId], 'district_id')
-
-      parcels[parcel.id].connectedTopLeft =
-        !parcels[topLeftId] ||
-        isSameValue(parcels[parcel.id], parcels[topLeftId], 'district_id')
+      parcels[id].connectedLeft = areConnected(parcels, id, leftId)
+      parcels[id].connectedTop = areConnected(parcels, id, topId)
+      parcels[id].connectedTopLeft = areConnected(parcels, id, topLeftId)
     }
   })
   return parcels
 }
 
+export function areConnected(parcels, parcelId, sideId) {
+  const parcel = parcels[parcelId]
+  const sideParcel = parcels[sideId]
+  return (
+    !sideParcel ||
+    isSameValue(parcel, sideParcel, 'district_id') ||
```

Comment:
I'm not convinced with this abstraction. It's not _really_ testing if two objects have the same value, it's testing that a property is not null in both _and_ it's the same.

I think it's only used here so I'm tempted to replace it with

```javascript
export function areConnected(parcels, parcelId, sideId) {
  const parcel = parcels[parcelId]
  const sideParcel = parcels[sideId]

  const sameDistrict = parcel.district_id === sideParcel.district_id
  const sameOwner = parcel.owner === sideParcel.ownwer

  return (
    !sideParcel ||
    (parcel.district_id && sameDistrict)
    (parcel.in_estate && parcel.owner && sameOwner) ||
  )
}
```

what do you think?

Reply:
Yes, my mistake was to force keep the `isSameValue`, I didn't even think if it was required or not, I'll put your implementation! (with typos fixed of course :P)